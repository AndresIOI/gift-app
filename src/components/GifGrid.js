import React from 'react';
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


const GifGrid = ({ category }) => {

  const {gifs, loading} = useFetchGifs(category);


  return (
    <>
      <h3>{ category }</h3>
      {
        loading ? 'Cargando...' : 
          (
            <div className="gif-grid animate__animated animate__fadeIn">
              {
                gifs.map( (gif) => (
                  <GifGridItem 
                    key={gif.id}
                    {...gif} 
                  />
                ))    
              }
            </div>
          )
      }
      
    </>
  )
}

export default GifGrid;