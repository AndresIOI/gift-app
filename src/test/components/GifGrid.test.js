import React from 'react';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en componente <GifGrid />', () => {
  const category = "Overwatch";

  test('debe de mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
      gifs: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect( wrapper ).toMatchSnapshot();

  });

  test('debe de mostrar items cuando se hace el fetch a useFetchGifs', () => {
    const gifs = [{
      id: "1",
      title: "title",
      img: "https://google.mx/img.jpg"
    },
    {
      id: "123",
      title: "title",
      img: "https://google.mx/img.jpg"
    }];

    useFetchGifs.mockReturnValue({
      gifs: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect( wrapper ).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect( wrapper.find('GifGridItem').length).toBe(gifs.length)

  });
})