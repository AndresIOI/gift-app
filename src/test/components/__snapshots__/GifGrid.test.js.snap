// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid /> debe de mostrar items cuando se hace el fetch a useFetchGifs 1`] = `
<Fragment>
  <h3>
    Overwatch
  </h3>
  <div
    className="gif-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="1"
      img="https://google.mx/img.jpg"
      key="1"
      title="title"
    />
    <GifGridItem
      id="123"
      img="https://google.mx/img.jpg"
      key="123"
      title="title"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Overwatch
  </h3>
  Cargando...
</Fragment>
`;
